### PR TITLE
feat(docs): add vite getting started docs

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
This PR adds a Vite getting started guide for React users.

#### Issue #, if available
Closes https://github.com/aws-amplify/amplify-ui/issues/3231

#### Description of how you validated changes
Smoke tested the docs:
https://reesscot-add-vite-docs.d22cn4asamk913.amplifyapp.com/react/getting-started/usage/vite

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/getting-started/usage/shared/BasicShoppingCard.tsx [hunk 1]
```diff
@@ -10,8 +10,8 @@ export const BasicShoppingCard = () => (
     <Image
       alt="Abstract art"
       height="21rem"
+      objectFit="cover"
       src="https://images.unsplash.com/photo-1500462918059-b1a0cb512f1d?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=987"
-      width="100%"
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/usage/shared/BasicShoppingCard.tsx:14]: This fixes and issue where the image renders skinnier than the actual image aspect ratio.

File: docs/src/pages/[platform]/getting-started/usage/vite/react.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+<ExampleCode>
+
+```tsx file=./examples/paintings.ts
+
+```
+
+  </ExampleCode>
+
+Then, replace the contents of `App.js` or `App.tsx` with the code below, restart the app, and experience all the beautiful components in action!
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/usage/vite/react.mdx:67]:
```suggestion
Then, replace the contents of `App.jsx` or `App.tsx` with the code below, restart the app, and experience all the beautiful components in action!
```

File: docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ]
+  }
+})
+```
+
+**3.** (Optional for TS users) If you run into TypeScript strict errors from `aws-amplify` you may need to update the `tsconfig.json` file with the following config:
+
+```json
+  "compilerOptions": {
+    "skipLibCheck": true,
```
Comment (0618) [inline docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx:35]: I think `skipLibCheck` is auto-generated by the template, but we do need to add `allowJs: true`

File: docs/src/data/links.tsx [hunk 4]
```diff
@@ -543,14 +543,20 @@ export const gettingStarted: ComponentNavItem[] = [
...
-    href: '/getting-started/usage/nextjs',
-    label: 'Next.js',
+    href: '/getting-started/usage/vite',
+    label: 'Vite',
+    platforms: ['react'],
+    tertiary: true,
+  },
+  {
+    href: '/getting-started/usage/create-react-app',
+    label: 'Create React App',
```
Comment (reesscot) [inline docs/src/data/links.tsx:559]: Moved CRA to the bottom, and Next to the top

File: docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx (hunk 3 above)
Comment (reesscot) [reply docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx:35]: It's autogenerated and set to `false`, which is why the workaround for `aws-amplify` TS errors requires setting this to `true`.

**Review by 0618 — APPROVED** [review]
left a nit comment

File: docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  }
+...
+```
+
+If you are still having issues, please see comments on the following issue for additional [Vite workarounds](https://github.com/aws-amplify/amplify-js/issues/9639). Note that there is active ongoing work to make these modifications unnecessary.
```
Comment (calebpollman) [inline docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx:40]: Maybe we can make this is a blockquote?
```suggestion
> If you are still having issues, please see comments on the following issue for additional [Vite workarounds](https://github.com/aws-amplify/amplify-js/issues/9639). Note that there is active ongoing work to make these modifications unnecessary.
```

**Review by calebpollman — APPROVED** [review]
🚢

File: docs/src/pages/[platform]/getting-started/usage/vite/react.mdx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+<Fragment>
+  {({ platform }) => import(`../../troubleshooting/shared/vite-polyfill.mdx`)}
+</Fragment>
+
+### Basic Demo
+
+Open up the project in your IDE and delete the contents of the `index.css` file. 
+
+Next replace the contents of the `App.jsx` or `App.tsx` file with the code below:
```
[the remark this answers is not in the document]
Comment (reesscot) [reply docs/src/pages/[platform]/getting-started/usage/vite/react.mdx:40]: The file extension is supposed to be `jsx` not `js`, so I'm confused by the suggestion

[the remark this answers is not in the document]
Comment (0618) [reply docs/src/pages/[platform]/getting-started/usage/vite/react.mdx:40]: Sorry, I mis-remembered it with CRA

File: docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx (hunk 3 above)
Comment (0618) [reply docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx:35]: Would you double check that? I just created a new app and got `"skipLibCheck": true,`

**Review by zchenwei — DISMISSED** [review]

**Review by calebpollman — APPROVED** [review]

File: docs/src/pages/[platform]/getting-started/usage/vite/react.mdx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+import { InstallScripts, TerminalCommand } from '@/components/InstallScripts.tsx';
+import { Fragment } from '@/components/Fragment';
+
+## Tutorial
+
+In this brief tutorial, we're going to build a basic shopping card component using [Vite.js](https://vitejs.dev/) and Amplify UI. Then, we'll elaborate on our shopping card to play with the possibilities of using Amplify UI.
+
+### Setup and Installation
+
+First, execute the command below in your terminal. When prompted for the name of your project, enter `amplify-ui-demo`.  Also see [Vite Guide](https://vitejs.dev/guide/)
```
Comment (wlee221) [inline docs/src/pages/[platform]/getting-started/usage/vite/react.mdx:13]:
```suggestion
First, execute the command below in your terminal. When prompted for the name of your project, enter `amplify-ui-demo`.  Also see [Vite Guide](https://vitejs.dev/guide/).

```

File: docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx (hunk 3 above)
Comment (reesscot) [reply docs/src/pages/[platform]/getting-started/troubleshooting/shared/vite-polyfill.mdx:35]: You're right, I'm seeing that now as well. I guess we don't need this anymore, I'll remove it now!

**Review by 0618 — APPROVED** [review]
🚢

**Review by zchenwei — APPROVED** [review]
LGTM

**Review by wlee221 — APPROVED** [review]

File: docs/src/pages/[platform]/getting-started/usage/vite/react.mdx [hunk 8]
```diff
@@ -0,0 +1,79 @@
+import { Tabs, TabItem } from '@aws-amplify/ui-react';
+import { Example, ExampleCode } from '@/components/Example';
+import { BasicShoppingCard, AdvancedShoppingCard } from '../shared';
+import { InstallScripts, TerminalCommand } from '@/components/InstallScripts.tsx';
+import { Fragment } from '@/components/Fragment';
+
+## Tutorial
+
+In this brief tutorial, we're going to build a basic shopping card component using [Vite.js](https://vitejs.dev/) and Amplify UI. Then, we'll elaborate on our shopping card to play with the possibilities of using Amplify UI.
```
Comment (kevinold) [inline docs/src/pages/[platform]/getting-started/usage/vite/react.mdx:9]: @reesscot Update "card" to "cart".

**Pull request merged**